mmv1/third_party/terraform/services/resourcemanager/data_source_google_client_config.go

```diff
@@ -38,14 +38,14 @@ type GoogleClientConfigModel struct {
 	DefaultLabels types.Map    `tfsdk:"default_labels"`
 }
 
-func (m *GoogleClientConfigModel) GetLocationDescription(providerConfig *fwtransport.FrameworkProviderConfig) fwresource.LocationDescription {
+func (m *GoogleClientConfigModel) GetLocationDescription(providerConfig *transport_tpg.Config) fwresource.LocationDescription {
 	return fwresource.LocationDescription{
 		RegionSchemaField: types.StringValue("region"),
 		ZoneSchemaField:   types.StringValue("zone"),
 		ResourceRegion:    m.Region,
 		ResourceZone:      m.Zone,
-		ProviderRegion:    providerConfig.Region,
-		ProviderZone:      providerConfig.Zone,
+		ProviderRegion:    types.StringValue(providerConfig.Region),
```

Comment:
Adding `types.StringValue()` is necessary because we're now directly accessing the SDK value instead of a pre-converted framework value?

Would we get a build error without adding `types.StringValue()` here? Is that how we know we've wrapped all of the appropriate fields that would need this?

Reply:
It's only necessary here because [the struct expects all inputs to be in the new type system](https://github.com/GoogleCloudPlatform/magic-modules/blob/27812e087aaf4250c076b5d572b3934c1a013e2e/mmv1/third_party/terraform/fwresource/framework_location.go#L16-L30). This is arbitrary - we could change `type LocationDescription struct` to expect type `string` values for ProviderRegion and ProviderZone and update the logic inside to check if the values `!= ""` etc.

This is an example of where we decide where and how the two type systems should meet. Should we enforce that all inputs to the `type LocationDescriber interface` logic are in the new type system and the responsibility for knowing about both type systems is with the calling code? Or do we write `type LocationDescriber interface` logic to handle both type systems?

Reply:
Got it, since that's the expected type system in the struct itself, the provider wouldn't build if a `string` value was attempted for those fields. 

> Should we enforce that all inputs to the type LocationDescriber interface logic are in the new type system and the responsibility for knowing about both type systems is with the calling code? Or do we write type LocationDescriber interface logic to handle both type systems?

I like keeping enforcing all inputs to be the new type system so that the writer has to be aware of the new type system, and it's not hidden in the interface logic. At least for now.